apistar_prometheus/components.py

```diff
@@ -53,8 +54,14 @@ class PrometheusHooks:
     def on_request(self, prometheus: Prometheus, method: http.Method, route: Route) -> None:
         prometheus.track_request_start(method, route and route.handler)
 
-    def on_response(self, prometheus: Prometheus, method: http.Method, route: Route, response: http.Response) -> None:
-        prometheus.track_request_end(method, route and route.handler, response)
+    def on_response(self, prometheus: Prometheus, method: http.Method, route: Route, response: http.Response, exc: Exception) -> None:
+        if exc is None:
+            prometheus.track_request_end(method, route and route.handler, response.status_code)
+        else:
```

Comment:
You should flatten this:

```python
if exc is None:
  ...
elif isinstance(ex, exceptions.HTTPException):
  ...
else:
  ...
```

Reply:
After checking with API Star docs, the last case isn't reachable, if the exception raised isn't a subclass of HTTPException it should be catched within `on_error`.